### PR TITLE
slender beam

### PR DESCRIPTION
**Pull request by rukawaaaaaaa**
add slender beam

### CODE REVIEW

**Comment by Xiangyu-Hu** [comment]
@rukawaaaaaaa what is the reason that the ci can not pass?

File: src/for_2D_build/common/data_type.h [hunk 1]
```diff
@@ -60,6 +60,7 @@ const Matd reduced_unit_matrix{
 
 /** initial local normal, only works for thin structure dynamics. */
 const Vecd local_pseudo_n_0 = Vecd(0.0, 1.0);
+const Vecd local_pseudo_b_n_0 = Vecd(0.0, 1.0);
```
Comment (Xiangyu-Hu) [inline src/for_2D_build/common/data_type.h:63]: As you code is only for 3d, it seems that you do need this.

**Comment by Xiangyu-Hu** [comment]
@rukawaaaaaaa when you initialize a new action by push, you may cancel the action which is still running.

**Comment by Xiangyu-Hu** [comment]
@rukawaaaaaaa Please have the local test (run all tests) 
first before you push to github. Otherwise it will lead to the overload of the actions.

**Comment by rukawaaaaaaa** [comment]
> @rukawaaaaaaa Please have the local test (run all tests) first before you push to github. Otherwise it will lead to the overload of the actions.

I am sorry, I will do the local test to avoid the overload.

File: src/shared/particle_generator/base_particle_generator.cpp
Comment (Xiangyu-Hu) [inline src/shared/particle_generator/base_particle_generator.cpp:1]: better put into for_3d_building?

File: src/shared/particles/solid_particles.cpp
Comment (Xiangyu-Hu) [inline src/shared/particles/solid_particles.cpp:1]: new particle class put into for_3d_build

File: src/shared/particles/solid_particles.h
Comment (Xiangyu-Hu) [inline src/shared/particles/solid_particles.h:1]: I think that you can move all the beam related class, including particles, particle generators.

**Review by Xiangyu-Hu — APPROVED** [review]

**Pull request merged**